sdk/automanage/Azure.ResourceManager.Automanage/src/Generated/Extensions/AutomanageExtensions.cs

```diff
@@ -0,0 +1,446 @@
+        }
+
+        /// <summary>
+        /// Retrieve a list of reports within a given configuration profile assignment
+        /// Request Path: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureStackHci/clusters/{clusterName}/providers/Microsoft.Automanage/configurationProfileAssignments/{configurationProfileAssignmentName}/reports
+        /// Operation Id: HCIReports_ListByConfigurationProfileAssignments
+        /// </summary>
+        /// <param name="resourceGroupResource"> The <see cref="ResourceGroupResource" /> instance the method will execute against. </param>
+        /// <param name="clusterName"> The name of the Arc machine. </param>
+        /// <param name="configurationProfileAssignmentName"> The configuration profile assignment name. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentException"> <paramref name="clusterName"/> or <paramref name="configurationProfileAssignmentName"/> is an empty string, and was expected to be non-empty. </exception>
+        /// <exception cref="ArgumentNullException"> <paramref name="clusterName"/> or <paramref name="configurationProfileAssignmentName"/> is null. </exception>
+        /// <returns> A collection of <see cref="Report" /> that may take multiple service requests to iterate over. </returns>
+        public static Pageable<Report> GetHCIReportsByConfigurationProfileAssignments(this ResourceGroupResource resourceGroupResource, string clusterName, string configurationProfileAssignmentName, CancellationToken cancellationToken = default)
+        {
+            Argument.AssertNotNullOrEmpty(clusterName, nameof(clusterName));
+            Argument.AssertNotNullOrEmpty(configurationProfileAssignmentName, nameof(configurationProfileAssignmentName));
+
+            return GetExtensionClient(resourceGroupResource).GetHCIReportsByConfigurationProfileAssignments(clusterName, configurationProfileAssignmentName, cancellationToken);
+        }
+
+        private static ArmResourceExtensionClient GetExtensionClient(ArmResource armResource)
+        {
+            return armResource.GetCachedClient((client) =>
+            {
+                return new ArmResourceExtensionClient(client, armResource.Id);
+            }
+            );
+        }
+
+        /// <summary> Gets a collection of ConfigurationProfileAssignmentResources in the ArmResource. </summary>
+        /// <param name="armResource"> The <see cref="ArmResource" /> instance the method will execute against. </param>
+        /// <returns> An object representing collection of ConfigurationProfileAssignmentResources and their operations over a ConfigurationProfileAssignmentResource. </returns>
+        public static ConfigurationProfileAssignmentCollection GetConfigurationProfileAssignments(this ArmResource armResource)
```

Comment:
Are we 100% sure this method applies to all resources?

There were some changes to how this is generated and your branch is using an [old version of autorest](https://github.com/AndrewCS149/azure-sdk-for-net/blob/andrsmith/automanageSDKTests/eng/Packages.Data.props#L163).  Sync from main and regenerate.  The default is now to take the id of the resource target as an extension off of ArmClient.  If this truly should apply to ALL resources then there is a configuration to generate it as it shows now which is an extension off of ArmResource.

Reply:
Thanks for pointing this out. Ive re-generated the SDK and modified the tests to support this.